src/main.cc

```diff
@@ -201,6 +203,18 @@ int main(int argc, char *argv[])
 
     app.init();
 
+    // Set system ID if specified via command line, for example --system-id:255
+    if (hasSystemId) {
+        bool ok;
+        int systemId = systemIdStr.toInt(&ok);
+        if (ok && systemId >= 0 && systemId <= 255) {  // MAVLink system IDs are 8-bit
+            qDebug() << "Setting MAVLink System ID to:" << systemId;
+            MAVLinkProtocol::instance()->setSystemId(systemId);
+        } else {
+            qDebug() << "Not setting MAVLink System ID. It must be between 0 and 255. Invalid system ID value:" << systemIdStr;
```

Comment:
I think this should be between 1 and 255. 0 is not a valid system ID in my understand as 0 means "all systems".

Reply:
You are right, thanks for spotting it! Fixed in https://github.com/mavlink/qgroundcontrol/pull/12410/commits/a7f460c700b35c86bae6febd9667f144f742e37f